cmd/config.go

```diff
@@ -166,6 +167,55 @@ func (t *TLSConfig) Load() (*tls.Config, error) {
 		return nil, fmt.Errorf("loading key pair from %q and %q: %s",
 			*t.CertFile, *t.KeyFile, err)
 	}
+
+	// tlsNotBefore is a prometheus gauge that outputs the TLS certificate's
+	// NotBefore field and registers it.
+	tlsNotBefore := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "tlsconfig_notbefore_seconds",
+			Help: "TLS certificate NotBefore field expressed as Unix epoch time",
+		},
+		[]string{"serial"})
+	err = scope.Register(tlsNotBefore)
+	if err != nil {
+		are := prometheus.AlreadyRegisteredError{}
+		if errors.As(err, &are) {
+			tlsNotBefore = are.ExistingCollector.(*prometheus.GaugeVec)
+		} else {
+			return &tls.Config{}, err
+		}
+	}
+
+	// tlsNotAfter is a prometheus gauge that outputs the TLS certificate's
+	// NotAfter field and registers it.
+	tlsNotAfter := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "tlsconfig_notafter_seconds",
+			Help: "TLS certificate NotAfter field expressed as Unix epoch time",
+		},
+		[]string{"serial"})
+	err = scope.Register(tlsNotAfter)
+	if err != nil {
+		are := prometheus.AlreadyRegisteredError{}
+		if errors.As(err, &are) {
+			tlsNotAfter = are.ExistingCollector.(*prometheus.GaugeVec)
+		} else {
+			return &tls.Config{}, err
+		}
+	}
+
+	leaf, err := x509.ParseCertificate(cert.Certificate[0])
+	if err != nil {
+		return &tls.Config{}, err
+	}
+	// There's a potential cardinality issue here where if we start using more
+	// TLS certificates or replacing them more often, we could degrade
+	// prometheus performance.
+	serial := (leaf.SerialNumber).String()
```

Comment:
I don't think these parens are necessary?
```suggestion
	serial := leaf.SerialNumber.String()
```